x/eibc/types/fulfill_order_authorization.go

```diff
@@ -0,0 +1,187 @@
+package types
+
+import (
+	"slices"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/authz"
+)
+
+// NewFulfillOrderAuthorization creates a new FulfillOrderAuthorization object.
+func NewFulfillOrderAuthorization(
+	rollapps []string,
+	denoms []string,
+	minFee sdk.IntProto,
+	maxPrice sdk.Coins,
+	fulfillerFeePart sdk.DecProto,
+	settlementValidated bool,
+	spendLimit sdk.Coins,
+) *FulfillOrderAuthorization {
+	return &FulfillOrderAuthorization{
+		Rollapps:            rollapps,
+		Denoms:              denoms,
+		MinFee:              minFee,
+		MaxPrice:            maxPrice,
+		FulfillerFeePart:    fulfillerFeePart,
+		SettlementValidated: settlementValidated,
+		SpendLimit:          spendLimit,
+	}
+}
+
+// MsgTypeURL implements Authorization.MsgTypeURL.
+func (a FulfillOrderAuthorization) MsgTypeURL() string {
+	return "/dymensionxyz.dymension.eibc.MsgFulfillOrderAuthorized"
+}
+
+// Accept implements Authorization.Accept.
+func (a FulfillOrderAuthorization) Accept(
+	ctx sdk.Context,
+	msg sdk.Msg,
+) (authz.AcceptResponse, error) {
+	mFulfill, ok := msg.(*MsgFulfillOrderAuthorized)
+	if !ok {
+		return authz.AcceptResponse{},
+			errorsmod.Wrapf(errors.ErrInvalidType,
+				"type mismatch: expected %T, got %T",
+				&MsgFulfillOrderAuthorized{}, msg)
+	}
+
+	// Check if the settlement validation flag matches
+	if a.SettlementValidated != mFulfill.SettlementValidated {
+		return authz.AcceptResponse{},
+			errorsmod.Wrapf(errors.ErrUnauthorized, "settlement validation flag mismatch")
+	}
+
+	// Check if the fulfiller fee part matches
+	if !a.FulfillerFeePart.Dec.Equal(mFulfill.FulfillerFeePart.Dec) {
+		return authz.AcceptResponse{},
+			errorsmod.Wrapf(errors.ErrUnauthorized, "fulfiller fee part mismatch")
+	}
+
+	// Check if the rollapp is allowed
+	if len(a.Rollapps) > 0 && !slices.Contains(a.Rollapps, mFulfill.RollappId) {
+		return authz.AcceptResponse{},
+			errorsmod.Wrapf(errors.ErrUnauthorized, "rollapp %s is not authorized", mFulfill.RollappId)
+	}
+
+	// Check if the denom is allowed
+	if len(a.Denoms) > 0 {
+		for _, orderDenom := range mFulfill.Price.Denoms() {
+			if !slices.Contains(a.Denoms, orderDenom) {
+				return authz.AcceptResponse{},
+					errorsmod.Wrapf(errors.ErrUnauthorized, "denom %s is not authorized", orderDenom)
+			}
+		}
+	}
+
+	// Check if the order fee meets the minimum fee
+	orderFee, ok := sdk.NewIntFromString(mFulfill.ExpectedFee)
+	if !ok {
+		return authz.AcceptResponse{},
+			errorsmod.Wrapf(errors.ErrInvalidCoins, "invalid fee amount")
+	}
+	if orderFee.LT(a.MinFee.Int) {
```

Comment:
seems like minFee is the actual amount and not in percentage? if so,  how does it fit with the liquidity provider specifying it in percentage?

Reply:
I thought it was agreed `minFee` and `maxPrice` ?